src/main/java/net/helix/pendulum/service/API.java

```diff
@@ -665,6 +666,106 @@ private AbstractResponse getNodeAPIConfigurationStatement() {
         return GetNodeAPIConfigurationResponse.create(configuration);
     }
 
+    /**
+     * <p>
+     *     Get the confirmation state of a set of transactions
+     *     This is for determining if a transaction was accepted and confirmed, i.e. finalized by the network.
+     * </p>
+     * <p>
+     *     This API call returns a list of boolean values in the same order as the submitted transactions.<br/>
+     *     Boolean values will be <tt>true</tt> for confirmed transactions, otherwise <tt>false</tt>.
+     * </p>
+     * Returns an {@link net.helix.pendulum.service.dto.ErrorResponse} if a transaction does not exist.
+     *
+     * @param transactions List of transactions that confirmation state is requested from
+     * @return {@link net.helix.pendulum.service.dto.GetInclusionStatesResponse}
+     * @throws Exception When a transaction cannot be loaded from hash
+     **/
+    private AbstractResponse getConfirmationStatesStatement(final List<String> transactions) throws Exception {
+        final List<Hash> trans = transactions.stream()
+                .map(HashFactory.TRANSACTION::create)
+                .collect(Collectors.toList());
+
+        int numberOfNonMetTransactions = trans.size();
+        final byte[] confirmationStates = new byte[numberOfNonMetTransactions];
+        int count = 0;
+        double threshold = BasePendulumConfig.Defaults.CONFIRMATION_THRESHOLD;
+        int n = BasePendulumConfig.Defaults.NUMBER_OF_ACTIVE_VALIDATORS;
+
+        for(Hash hash: trans) {
+            TransactionViewModel transaction = TransactionViewModel.fromHash(tangle, hash);
+
+            log.debug("tx_confirmations {}:[{}:{}]", transaction.getHash().toString(), transaction.getConfirmations(), (double) transaction.getConfirmations() / n);
+
+            // is transaction finalized
+            if(((double)transaction.getConfirmations() / n) > threshold) {
+                confirmationStates[count] = 1;
+            }
+            // not finalized yet
+            else {
+                confirmationStates[count] = -1;
+            }
+            count++;
+        }
+
+
+        final boolean[] confirmationStatesBoolean = new boolean[confirmationStates.length];
+        for(int i = 0; i < confirmationStates.length; i++) {
+            // If a state is 0 by now, we know nothing so assume not included
+            confirmationStatesBoolean[i] = confirmationStates[i] == 1;
+        }
+        return GetInclusionStatesResponse.create(confirmationStatesBoolean);
```

Comment:
GetInclusionStatesResponse should also be renamed

Reply:
> GetInclusionStatesResponse should also be renamed

The renaming I will do, when I rename command and adapt SDK/Wallet.